cli/program_state.rs

```diff
@@ -56,7 +56,7 @@ pub struct ProgramState {
 }
 
 impl ProgramState {
-  pub fn new(flags: flags::Flags) -> Result<Arc<Self>, AnyError> {
+  pub async fn new(flags: flags::Flags) -> Result<Arc<Self>, AnyError> {
```

Comment:
Maybe `ProgramState::new` should be renamed to `ProgramState::build` because it is now async?